ui/widgets/sidebar/src/components/sidebar-component.tsx

```diff
@@ -203,10 +189,18 @@ const SidebarComponent: React.FC<unknown> = () => {
   }
 
   async function handleLogout() {
-    await logoutQuery.mutateAsync();
-    await queryClient.invalidateQueries({
-      queryKey: [LOGIN_STATE_KEY],
+    setIsLoading(true);
+
+    queryClient.resetQueries({
+      queryKey: useGetMyProfileQuery.getKey(),
+    });
+    await logoutQuery.mutateAsync().then(() => {
```

Comment:
this should make use of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race

Reply:
the list for race contains the `mutateAsync` and the timeout functions

Reply:
the idea is that this should not get stuck if the logoutQuery does not respond